packages/sdk/src/wallet/core/wallets/smart/default/DefaultSmartWallet.ts

```diff
@@ -222,6 +239,99 @@ export class DefaultSmartWallet extends SmartWallet {
     }
   }
 
+  /**
+   * Add a new signer to the smart wallet
+   * @description Adds either an EOA address signer or a WebAuthn account signer
+   * to the underlying smart wallet contract. For WebAuthn accounts, the method
+   * extracts the x and y coordinates from the provided 64-byte public key and
+   * calls the contract's `addOwnerPublicKey`. For EOA addresses it calls
+   * `addOwnerAddress`. The add operation is sent as a UserOperation via
+   * {@link sendBatch}, and upon success the method queries the contract to
+   * resolve the signer's index.
+   * @param signer - Ethereum address (EOA) or a `WebAuthnAccount` to add
+   * @param chainId - Target chain on which the smart wallet operates
+   * @returns Promise resolving to the onchain signer index for the newly added signer
+   * @throws Error if the add operation fails or the owner index cannot be found
+   */
+  async addSigner(
+    signer: Address | WebAuthnAccount,
+    chainId: SupportedChainId,
+  ): Promise<number> {
+    const calls = []
+    if (typeof signer === 'string') {
+      calls.push({
+        to: this.address,
+        data: encodeFunctionData({
+          abi: smartWalletAbi,
+          functionName: 'addOwnerAddress',
+          args: [signer] as const,
+        }),
+        value: toHex(0),
+      })
+    } else if (signer.type === 'webAuthn') {
+      const [x, y] = decodeAbiParameters(
+        [{ type: 'bytes32' }, { type: 'bytes32' }],
+        signer.publicKey,
+      )
+      calls.push({
+        to: this.address,
+        data: encodeFunctionData({
+          abi: smartWalletAbi,
+          functionName: 'addOwnerPublicKey',
+          args: [x, y] as const,
+        }),
+        value: toHex(0),
+      })
+    }
+
+    const receipt = await this.sendBatch(calls, chainId)
+
+    if (!receipt.success) {
+      throw new Error('add signer call failed')
+    }
+
+    const publicKey = typeof signer === 'string' ? signer : signer.publicKey
+    let signerIndex = await findOwnerIndex({
+      address: this.address,
+      publicKey,
+      client: this.chainManager.getPublicClient(chainId),
+    })
+    // sometimes the index is not found immediately after the add owner call
+    // so we wait 2 seconds and try again.
```

Comment:
Do we have similar "wait and try again" logic anywhere?  Whenever we have it in a few places, it would be good to pull into a util